test/unit/redux/studio-member-actions.test.js

```diff
@@ -0,0 +1,377 @@
+    });
+
+    test('it correctly uses the admin route when possible', () => {
+        store = configureStore(reducers, {
+            ...initialState,
+            studio: {id: 123123},
+            session: {
+                session: {
+                    user: {token: 'a-token'},
+                    permissions: {admin: true}
+                }
+            }
+        });
+        api.mockImplementation((opts, callback) => {
+            expect(opts.uri).toBe('/admin/studios/123123/curators/');
+            expect(opts.authentication).toBe('a-token');
+        });
+        store.dispatch(loadCurators());
+    });
+
+    test('errors are set on the curators state', () => {
+        store = configureStore(reducers, {
+            ...initialState,
+            studio: {id: 123123}
+        });
+        api.mockImplementation((opts, callback) => {
+            callback(null, null, {statusCode: 500});
+        });
+        store.dispatch(loadCurators());
+        expect(curators.selector(store.getState()).error).toBe(Errors.SERVER);
+    });
+});
+
+describe('removeManager', () => {
+    beforeEach(() => {
+        store = configureStore(reducers, {
+            ...initialState,
+            studio: {
+                id: 123123,
+                managers: 3,
+                manager: true
+            },
+            managers: {
+                items: [
+                    {username: 'user1'},
+                    {username: 'user2'},
+                    {username: 'user3'}
+                ]
+            },
+            session: {
+                session: {
+                    user: {username: 'user2'}
+                }
+            }
+        });
+    });
+
+    test('removes the manager by username and decrements the count', async () => {
+        api.mockImplementation((opts, callback) => {
+            expect(opts.uri).toBe('/site-api/users/curators-in/123123/remove/');
+            callback(null, {}, {statusCode: 200});
+        });
+
+        await store.dispatch(removeManager('user2'));
+        const state = store.getState();
+
+        // Ensure it removes the correct manager (index=1)
+        expect(selectStudioManagerCount(state)).toBe(2);
+        expect(managers.selector(state).items[0].username).toBe('user1');
+        expect(managers.selector(state).items[1].username).toBe('user3');
+
+        // Ensure roles change if you are removing yourself
+        expect(state.studio.manager).toBe(false);
+    });
+
+    test('on error, promise rejects without any changing count or list', async () => {
+        api.mockImplementation((opts, callback) => {
+            callback(null, {}, {statusCode: 403});
+        });
+
+        await expect(store.dispatch(removeManager('user2')))
+            .rejects.toBe(Errors.PERMISSION);
+
+        const state = store.getState();
+        const {items} = managers.selector(state);
+        expect(selectStudioManagerCount(state)).toBe(3);
+        expect(items.length).toBe(3);
+    });
+});
```

Comment:
Thats a really good point, I'll update the test (and the bug you pointed out in the code). 

One way to be better about covering those callbacks is to use code coverage stats. I tried adding `--coverage` while running this test and it pointed out some of these untested `ifs` in the callback, but the reporter was difficult to understand so I may look for different command line options for it. I know we used to (perhaps still do?) use code coverage stats with some hosted CI-integrated service, and that would visually show which lines of your source werent covered when you submitted PRs.

Reply:
Actually I think the functionality is correct, if the server says "ok i removed the manager" and the frontend says "i don't have that manager in my list", then the frontend should still probably decrement the manager count, because the frontend list of managers is not authoritative (its paginated for one, and may be out of date for other reasons). I'll at least document this in the code. Of course the frontends manager count can also be out-of-date for other reasons (not pagination, but e.g. other managers have promoted other managers while you were on the page). I guess ideally the manager removal endpoint would respond with the current number of managers to keep things more in sync...